### PR TITLE
Check cidr value

### PR DESCRIPTION
**Pull request by hmunfru**
@jesuspg @jframos

### CODE REVIEW

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 1]
```diff
@@ -414,10 +414,7 @@ private String getDefaultCidr(ClaudiaData claudiaData, String region) throws Inv
         if (!(claudiaData.getVdc() == null || claudiaData.getVdc().isEmpty())) {
             cidrOpenstack = getNumberDeployedNetwork(claudiaData, region) + 1;
         }
-
-        int cidrdb = this.findAll().size();
-        int cidrCount = cidrdb + cidrOpenstack;
-        return "10.0." + cidrCount + ".0/24";
+        return "10.1." + cidrOpenstack + ".0/24";
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: Magic Number...

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 2]
```diff
@@ -414,10 +414,7 @@ private String getDefaultCidr(ClaudiaData claudiaData, String region) throws Inv
         if (!(claudiaData.getVdc() == null || claudiaData.getVdc().isEmpty())) {
             cidrOpenstack = getNumberDeployedNetwork(claudiaData, region) + 1;
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: You have to explain why are you adding 1 to the getNumberDeployedNetwork.

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 3]
```diff
@@ -255,11 +258,11 @@ public void delete(ClaudiaData claudiaData, NetworkInstance networkInstance, Str
         }
 
     }
-    
+
     private void deleteInDb(NetworkInstance networkInstance) throws InvalidEntityException {
     log.info("Destroying network in DB" + networkInstance.getNetworkName() + " " + networkInstance.getSubNets().size() );
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: I see incorrect indentation of this method.

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 4]
```diff
@@ -347,31 +351,34 @@ private NetworkInstance createInBD (ClaudiaData data, NetworkInstance netInstanc
...
         return netInstance;
-
     }
-    
-    private boolean existsInOpenstack (ClaudiaData claudiaData, NetworkInstance network, String region) {
+
+    private boolean existsInOpenstack (ClaudiaData claudiaData,
+                                       NetworkInstance network, String region) {
     	try {
 			networkClient.loadNetwork(claudiaData, network, region);
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: check style

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 5]
```diff
@@ -395,29 +403,78 @@ public void setSubNetworkInstanceManager(SubNetworkInstanceManager subNetworkIns
...
+
+        for (SubNetworkInstance subNet: subNets) {
+            if (subNet.getCidr().equals(cidr)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public String getValidCidr (ClaudiaData claudiaData, String region,int subNetElement) throws InfrastructureException{
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: ClaudiaData claudiaData, String region, int subNetElement

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java [hunk 6]
```diff
@@ -395,29 +403,78 @@ public void setSubNetworkInstanceManager(SubNetworkInstanceManager subNetworkIns
...
+        String cidr = String.format(NETWORK_CIDR, NET_ELEMENT, subNetElement);
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: It is correct that you have always 10.2.xx.0? I mean always are you putting 2 in the second component of the IP...

Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: no. net_element can increase

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java (hunk 4 above)
Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: segun el coding style esta linea está ok.

Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/manager/impl/NetworkInstanceManagerImpl.java]: Show the content in github...

**Pull request closed without merging**